src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2315,7 +2315,8 @@ private static IEnumerable<Diagnostic> FilterDiagnosticsByLocation(IEnumerable<D
             if (stage == CompilationStage.Declare || (stage > CompilationStage.Declare && includeEarlierStages))
             {
                 var declarationDiagnostics = GetSourceDeclarationDiagnostics(syntaxTree, filterSpanWithinTree, FilterDiagnosticsByLocation, cancellationToken);
-                Debug.Assert(declarationDiagnostics.All(d => d.HasIntersectingLocation(syntaxTree, filterSpanWithinTree)));
+                // re-enable the assert once https://github.com/dotnet/roslyn/issues/21185 is fixed
+                // Debug.Assert(declarationDiagnostics.All(d => d.HasIntersectingLocation(syntaxTree, filterSpanWithinTree)));
```

Comment:
Note: this may be a dupe of https://github.com/dotnet/roslyn/issues/21020

Is this only reproing for the same case that cheryl was reporting?  Or is this for more cases?